powershell/tests/functions/Common.Tests.ps1

```diff
@@ -0,0 +1,56 @@
+﻿BeforeDiscovery {
+    $moduleName = 'Maester'
+    $moduleRoot = "$PSScriptRoot/../.."
+    # Get all the functions in the /public folder
+    $exportedFunctions = Get-Command -Module $moduleName -CommandType Function
+
+    # Eventually this should include all functions in the /public folder
+    # For now, just the ones that we have tested and added
+    $functionsWithTests = @('Invoke-Maester')
+}
+
+Describe 'Common function tests' -Tags 'Acceptance' -ForEach @{ exportedFunctions = $exportedFunctions; moduleRoot = $moduleRoot } {
+    Context '<_.CommandType> <_.Name>' -ForEach $exportedFunctions {
+        BeforeAll {
+            $function = $_
+            # Need to update this if we start building the module as a single psm1 file (for improved performance)
+            $functionPath = $_.ScriptBlock.File
+        }
+
+        It "<function>.ps1 should exist in public folder" {
+            # Normalize path in test to work cross-platform
+            $functionPath -replace '\\','/' | Should -BeLike "*/public/*$($function.Name).ps1"
+            $functionPath | Should -Exist
+        }
+
+        It "Should be an advanced function" {
+            $function.CmdletBinding | Should -BeTrue -Because 'public functions should be advanced functions'
+            $function.ScriptBlock.Ast.Body.ParamBlock | Should -Not -BeNullOrEmpty -Because 'functions should have a param()-block'
+            $function.ScriptBlock.Ast.Body.ParamBlock | Should -Not -BeNullOrEmpty -Because 'functions should have [CmdletBinding()] attribute for explicit advanced function'
+        }
+
+        # Skipping for Cisa tests until they're updated
```

Comment:
Would we need this for EIDSCA as well? Or is this not required because we only focus on public functions as of now?

Reply:
Public functions for now, so only applicable to `Test-MtEidscaControl` which is fixed.

Should probably create an issue so someone familiar with the Cisa functions can fix them.

Personally I'm not sure the test is needed, but kept it for now. It might force useless verbose output just to be compliant, e.g. `Get-MtSession`.